### PR TITLE
MNTOR-1306: Premium CTA/badge

### PR DESCRIPTION
**Pull request by flozia**
# References: 
Jira: [MNTOR-1306](https://mozilla-hub.atlassian.net/browse/MNTOR-1306)
Figma: https://www.figma.com/file/iJFOikscJeplRBOuKOtYEW/Onboarding?type=design&mode=design&t=WFSl7cGOyYsWtuTX-0

# Description

Adds the premium upsell CTA for free users and badge for premium users. The upsell dialog will be implemented in [MNTOR-1292](https://mozilla-hub.atlassian.net/browse/MNTOR-1292).

# Screenshot

| CTA | Badge |
| --- | --- |
| mntor-1306-cta | mntor-1306-badge |

# How to test

- Run Storybook: `npm run storybook`
- You can see the variants under here:
  - CTA: http://localhost:6006/?path=/story/pages-dashboard--dashboard-free-user
  - Badge: http://localhost:6006/?path=/story/pages-dashboard--dashboard-premium-user

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: locales-pending/premium.ftl [hunk 1]
```diff
@@ -26,6 +26,9 @@ toolbar-app-picker-by-mozilla = Made by { -brand-mozilla }
 footer-external-link-faq-label = FAQs
 footer-external-link-faq-tooltip = Frequently asked questions
 
+premium-badge-label = { -brand-premium }
+premium-cta-label = Upgrade to Premium
```
Comment (flodolo) [inline locales-pending/premium.ftl:30]:
```suggestion
premium-cta-label = Upgrade to { -brand-premium }
```

I assume we want to enforce the same brand use here?

Comment (flozia) [reply locales-pending/premium.ftl]: Right — updated!

**Review by flodolo — APPROVED** [review]

File: src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.test.tsx [hunk 2]
```diff
@@ -62,3 +64,23 @@ it("switches between tab panels", async () => {
   expect(tabFixedTrigger.getAttribute("aria-selected")).toBe("true");
   expect(tabActionNeededTrigger.getAttribute("aria-selected")).toBe("false");
 });
+
+it("shows the premium upgrade cta if the user is not a premium subscriber", () => {
+  const ComposedDashboard = composeStory(DashboardFreeUser, Meta);
+  render(<ComposedDashboard />);
+
+  // We show a CTA on desktop in the toolbar and in the mobile menu
```
Comment (Vinnl) [inline src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.test.tsx:72]: Explaining the `2` just where it's needed, nice :)

**Review by Vinnl — APPROVED** [review]
LGTM!

**Pull request merged**